pkg/authentication/externalidp/awsiam/aws_client.go

```diff
@@ -0,0 +1,153 @@
+package awsiam
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+	"github.com/google/uuid"
+)
+
+var ErrInvalidCredentialsFormat = errors.New("missing required parts in credentials")
+
+const (
+	AuthVersion       = "2011-06-15"
+	AuthMethod        = http.MethodPost
+	AuthAction        = "GetCallerIdentity"
+	AuthAlgorithm     = "AWS4-HMAC-SHA256"
+	StsGlobalEndpoint = "sts.amazonaws.com"
+	AuthActionKey     = "Action"
+	AuthVersionKey    = "Version"
+	AuthAlgorithmKey  = "X-Amz-Algorithm"
+	//nolint:gosec
+	AuthCredentialKey = "X-Amz-Credential"
+	AuthDateKey       = "X-Amz-Date"
+	AuthExpiresKey    = "X-Amz-Expires"
+	//nolint:gosec
+	AuthSecurityTokenKey  = "X-Amz-Security-Token"
+	AuthSignedHeadersKey  = "X-Amz-SignedHeaders"
+	AuthSignatureKey      = "X-Amz-Signature"
+	DatetimeFormat        = "20060102T150405Z"
+	CredentialTimeFormat  = "20060102"
+	DefaultSTSLoginExpire = 15 * time.Minute
+
+	mimLengthSplitCreds = 3
```

Comment:
Part of go and out go styling, encapsulate consts in logical groups, all of the above are query params, and those are just random consts 

```suggestion
const (
	DefaultSTSLoginExpire = 15 * time.Minute

	mimLengthSplitCreds = 3
)
```